### DISCUSSION TITLE
Moving cluster config files from one directory to another (same cluster, same configs, just different path)

### DISCUSSION THREAD

**Discussion by AnchorArray** [fluxcd/flux2#4053] [Q&A]
The documentation [How can I safely move resources from one dir to another?](https://fluxcd.io/flux/faq/#how-can-i-safely-move-resources-from-one-dir-to-another) states:

> To move manifests from a directory synced by a Flux Kustomization to another dir synced by a different Kustomization, first you need to disable garbage collection then move the files.

Some additional background we deploy our product in multiple environments and support single-tenant deployments where we deploy a dedicated infrastructure in a dedicated cloud provider account for each single-tenant customer. We have these in the same repository as we have a `stable` directory that includes the base configurations for all single tenants and use overlays to override the tenant specific configurations. This allows us to release new versions of our application by updating just the `stable` directory. 

All of our existing environments are organized near the top level of the repository. But, we want to nest all of our existing AWS accounts under an `aws` directory, and leave space for `gcp` and `azure` sub-directories in the future as these different clouds will have slightly different configurations. 

The documentation around `mov[ing] resources from one dir to another` feels like it's geared more toward moving resources from one "Flux Kustomization" (presumably the kustomization that stores the path within the repo that is synced with the cluster) to another. In this scenario, its still the same cluster, we're just changing the path.

The initial strategy we proposed was:

- Create a duplicate of the clusters directory nested under the desired cloud-provider, i.e. if we have `environments/integration/devops` we'll create a copy of the contents of this directory to `/environments/integration/aws/devops` (updating any path references) so that both exist at the same time but cluster won't be syncing with `/environments/integration/aws/devops` yet.
- Execute `flux bootstrap` with the new `--path` argument to point to the new `/environments/integration/aws/devops` directory with identical content from the previous path.
- Delete the previous non-cloud-specific directory.

I guess the primary question is if the documentation referenced above applies to the task that we are tackling, i.e. setting `prune` to `false`, or would the initial strategy outlined above be sufficient if it's the same configs for the same cluster, just changing the `--path` when running `flux bootstrap`?

Thanks in advance!!

**Comment by mahdisoussi** [comment]
Hello, any ideas on how to achieve this these days ? Or how did you do it ?

**Reply by stefanprodan** [reply]
The procedure described by @AnchorArray is correct.

**Comment by airtonix** [comment]
This didn't exactly work as @AnchorArray says... I had to perform a few extra steps: 

- clone the directory, with a new name.
- modify references in the new directory.
- commit and push this up.
- ⚠️ if you try to run bootstrap now, it will complain about not wanting to overwrite things. `what evs. don't care`
- `kubectl edit kustomization flux-system -n flux-system` and point the current cluster config path to the newly modified and pushed path.
- now, run `flux reconcile kustomization flux-system --with-source`
- confirm your cluster is being configured by your new directory.
- remove the old directory and push it up.
